src/Sarif.Sarifer/SariferPackage.cs

```diff
@@ -93,5 +110,24 @@ public void Dispose()
             this.Dispose(disposing: true);
             GC.SuppressFinalize(this);
         }
+
+        private async System.Threading.Tasks.Task<bool> IsSolutionLoadedAsync()
+        {
+            await this.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            IVsSolution solutionService = await this.GetServiceAsync(typeof(SVsSolution)).ConfigureAwait(continueOnCapturedContext: true) as IVsSolution
+                ?? throw new InvalidOperationException("Cannot load solution service.");
+
+            ErrorHandler.ThrowOnFailure(solutionService.GetProperty((int)__VSPROPID.VSPROPID_IsSolutionOpen, out object value));
+
+            return value is bool isSolutionOpen && isSolutionOpen;
+        }
+
+        private void OnSolutionLoaded(object sender = null, EventArgs e = null)
```

Comment:
>OnSolutionLoaded [](start = 21, length = 16)

What's going to happen here? I didn't ask for a background analysis on solution load. This could greatly slow things down after SLN load. Instead, the analyzer should inspect all opened files only, unless/until the user invokes a specific gesture to analyze the entire solution.

Reply:
Ok, I'll withdraw this.

---
In reply to: [532939181](https://github.com/microsoft/sarif-visualstudio-extension/pull/298#discussion_r532939181) [](ancestors = 532939181)